staff/sofia-costa/precourse/bingo/index.js

```diff
@@ -0,0 +1,311 @@
+
+var randomNumbers = []
+
+//Variable para generar 3 lineas:
+
+var card = [[], [], []];
+
+var cardComplete = [["X","X","X","X","X"], ["X","X","X","X","X"], ["X","X","X","X","X"]]
+
+var totalPoints = 0
+
+// Variable para averiguar se se hizo o no linea:
+
+var isLine = false
+
+alert('Welcome to Bingo! \n Powered by Skylab - Together We May Reach the Sky.')
+
+getName()
+
+
+
+
+// Funcion donde se obtiene el nombre del usuario:
+
+function getName() {
+    var whatName = prompt('What is your name?').toUpperCase()
+    if (whatName !== '') {nameUser.name = whatName
+                        return bingo()}
+    else if(whatName === '') {alert('Please introduce your name :)')
+                            return getName()}
+}
+
+
+
+
+// Funcion que mostra las reglas del juego. Se el usuario la acepta empieza el juego:
+
+function bingo () { 
+
+    var startConfirm = confirm(`Oh hi there ${nameUser.name}! Here are the rules of our bingo game:
+
+You are given a card with numbers. Random numbers will be shown, and if one of those matches a number \
+in your card it will be marked with a cross. When you make a line you will be noticed! \
+When you have all the numbers, you will get BINGO!
+
+The less attempts you need to do BINGO!, the more points you earn. \
+Your points will be accumulated in your user, and user punctuations will be compared.
+
+There you go! Shall we start?`)
+
+    if (startConfirm === true) {return getCard()}
+    else (startConfirm === false) ; {return alert('Well, you just lost the chance to play a pretty fun game ;) See you next time!')}
+
+}
+
+
+
+
+// Funcion para generar el carton del usuario. Se él acepta, se mantiene, se la rechaza, se genera una nueva:
+
+function getCard() {
+
+    randomNums()
+
+    // Aquí se añaden numeros random en linea que después serán convertidos en 3 lineas:
+    
+    function randomNums() {
+
+        var num = Math.floor((Math.random() * 90) + 1)
+
+        if (line.length === 0) {line.push(num)}
+        else if (line.length < 15) { line.includes(num) ? '' : line.push(num) }
+
+        if(line.length < 15) {randomNums()}
+        else if(line.length === 15) {threeLines()}
+
+    }  
+
+    // La conversion en 3 lineas:
+
+    function threeLines() {
+        for (var i in card) {
+            card[i] = line.splice(0,5)
+            }
+    }
+
+    console.log(
+`
+Line 1:   ${card[0].join('  ')}
+Line 2:   ${card[1].join('  ')}
+Line 3:   ${card[2].join('  ')}
+
+`)
+
+    // Para confirmar se el usuario quiere mantener su carton o generar un nuevo:
+            
+    var confirmBombo = confirm('Confirm to keep this sheet of number or cancel to get another one.')
+            
+        if(confirmBombo === true) {return showRandomNum()}
+        else (confirmBombo === false) ; {return getCard()}
+            
+}
+
+
+
+
+// Funcion para los numeros random que salen en las bolas:
+
+    
+function getRandomNumber() {
+
+    var num = Math.floor((Math.random() * 90) + 1)
```

Comment:
You can change .ceil by .floor and remove + 1 to the final, so the line always will be working.

Reply:
Nice advice! Thank you :)